pkg/artifact/bigquery.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2024 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package artifact
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"cloud.google.com/go/bigquery"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+)
+
+// BigQuery provides a client and dataset identifiers.
+type BigQuery struct {
+	projectID string
+	datasetID string
+	client    *bigquery.Client
+}
+
+// sourceQuery is the driving BigQuery query that selects events
+// that need to be processed.
+const SourceQuery = `
```

Comment:
Does this need to be public?

Reply:
No, not anymore. It did when it was in Beam for <insert unknown Beam madness reasons>.